app/routes/goal_routes.py

```diff
@@ -1 +1,107 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request, abort, make_response
+from app.models.goal import Goal
+from app.routes.task_routes import validate_task
+from ..db import db
+from app.models.task import Task
+
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+@goals_bp.post("")
+def create_goal():
+    request_body = request.get_json()
+    
+    try:
+        new_goal = Goal.from_dict(request_body)
+    except KeyError as error:
+        response = {"details": "Invalid data"}
+        abort(make_response(response, 400))
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal":new_goal.to_dict()}, 201
```

Comment:
We essentially just want an outer dictionary structure with a `key` of the class name, perhaps we could move this into a helper function or add it to a class in order to DRY up some of our code given that we have this same logic throughout the project?